modules/edu.gemini.seqexec.web/edu.gemini.seqexec.web.client/src/main/scala/edu/gemini/seqexec/web/client/components/sequence/steps/StepSettings.scala

```diff
@@ -245,16 +247,20 @@ object StepIdCell {
   * Component to link to the settings
   */
 object SettingsCell {
-  private val component = ScalaComponent.builder[Int]("SettingsCell")
+  final case class Props(ctl: RouterCtl[Pages.SeqexecPages], instrument: Instrument, obsId: SequenceId, index: Int)
+  private val component = ScalaComponent.builder[Props]("SettingsCell")
     .stateless
     .render_P { p =>
+      val page = Pages.SequenceConfigPage(p.instrument, p.obsId, p.index + 1)
       <.div(
         SeqexecStyles.settingsCell,
-        IconCaretRight.copyIcon(fitted = true)
+        p.ctl.link(page)(
+          IconCaretRight.copyIcon(color = "black".some, onClick = SeqexecCircuit.dispatchCB(NavigateSilentTo(page)))
```

Comment:
I think I understand what the end result of this is, but there is a lot happening here that I'm not certain of how it all works.

Reply:
Here we are creating a link to a page of the react router application. This lets us have meaningful urls even when it is a SPA

This is a relatively large subject
https://github.com/japgolly/scalajs-react/blob/master/doc/ROUTER.md